src/transformers/trainer.py

```diff
@@ -515,11 +514,6 @@ def evaluate(
         eval_dataloader = self.get_eval_dataloader(eval_dataset)
 
         output = self._prediction_loop(eval_dataloader, description="Evaluation")
-        if _has_wandb and not no_log and wandb.run:
-            # Log final metrics
-            wandb.log(
-                {**{"eval_{}".format(k): v for k, v in output.metrics.items()}, "epoch": self.args.num_train_epochs}
-            )
```

Comment:
If one needs this, probably simpler to just do it from the calling script when you have the eval results

Reply:
Makes sense + it simplifies the trainer

Reply:
(personally I always use the `--evaluate_during_training` flag so I don't really need it here anyways)

Reply:
I just remembered the use. It was too get final evaluation results at last training step (which may not be a logging step otherwise).

In those cases I works just log evaluation metrics (not the training ones).

On the runs I sent, we can see the metrics are logged at the end (but no training loss).

The idea is to have final performance on the model. What do you think?